lib/deployer.rb

```diff
@@ -56,7 +57,15 @@ def deploy_all
 
   private
 
-  def prompt_user_for_confirmation!
+  def prompt_user_for_main_confirmation!
+    return unless @tag.blank? || @tag != 'main'
+
+    confirmation = TTY::Prompt.new.yes?('You are deploying without a tag, which will deploy the main branch of all repos.  Are you sure?') { |prompt| prompt.default(false) }
+
+    abort 'Deployment to main aborted.' unless confirmation
+  end
+
+  def prompt_user_for_approval_confirmation!
     return if repos_requiring_confirmation.empty?
```

Comment:
renamed this method from `prompt_user_for_confirmation` to `prompt_user_for_approval_confirmation` to reduce ambiguity about the different types of confirmations required